diff -ruN kyotocabinet-1.2.76.orig/kcdbext.h kyotocabinet-1.2.76/kcdbext.h
--- kyotocabinet-1.2.76.orig/kcdbext.h	2012-05-24 17:27:59.000000000 +0100
+++ kyotocabinet-1.2.76/kcdbext.h	2014-02-18 14:38:31.322029834 +0000
@@ -1278,7 +1278,7 @@
     if (omode_ == 0) {
       set_error(_KCCODELINE_, BasicDB::Error::INVALID, "not opened");
       *sp = 0;
-      return false;
+      return NULL;
     }
     if (!cache_) return db_.get(kbuf, ksiz, sp);
     size_t dvsiz = 0;
